File: drestaurant-libs/drestaurant-common/src/main/java/com/drestaurant/common/domain/command/AuditableAbstractCommand.java
package com.drestaurant.common.domain.command;

import com.drestaurant.common.domain.model.AuditEntry;

/**
 *
 * @author: idugalic
 * Date: 4/29/18
 * Time: 4:03 PM
 */
public class AuditableAbstractCommand {

	private AuditEntry auditEntry;

	public AuditableAbstractCommand(AuditEntry auditEntry) {
		this.auditEntry = auditEntry;
	}

	public AuditEntry getAuditEntry() {
		return auditEntry;
	}
}

File: drestaurant-libs/drestaurant-core-api/src/main/kotlin/com/drestaurant/common/domain/api/AuditableAbstractCommand.kt
package com.drestaurant.common.domain.api

import com.drestaurant.common.domain.api.model.AuditEntry

/**
 * Abstract command
 *
 * @property auditEntry audit entry holds the information of 'who' and 'when' performed the command
 */
abstract class AuditableAbstractCommand(open val auditEntry: AuditEntry)
